### PR TITLE
Enable doctests, update documentation and enable pytest in travis.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -34,255 +34,34 @@
...
-# documentation.
-#html_theme_options = {}
-
-# Add any paths that contain custom themes here, relative to this directory.
-#html_theme_path = []
-
-# The name for this set of Sphinx documents.  If None, it defaults to
-# "<project> v<release> documentation".
-#html_title = None
+html_theme = 'alabaster'
```
Comment (pelson) [inline doc/source/conf.py:61]: This is a genuine change. It moves us from the default, to alabaster theme.

File: doc/source/conf.py [hunk 2]
```diff
@@ -34,255 +34,34 @@
...
-#html_use_opensearch = ''
-
-# This is the file name suffix for HTML files (e.g. ".xhtml").
-#html_file_suffix = None
-
-# Output file base name for HTML help builder.
-htmlhelp_basename = 'cf_unitsdoc'
-
-
-# -- Options for LaTeX output ---------------------------------------------
```
Comment (pelson) [inline doc/source/conf.py:207]: As far as I'm aware, we've never actually published a PDF manual. Given we don't for cartopy nor Iris, I don't think we need to do it for cf_units...

**Comment by pelson** [comment]
I'd like to see the coveralls response on this PR, but otherwise, I think its ready.

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
-coveralls
```
Comment (bjlittle) [inline requirements-dev.txt]: @pelson Don't we need to keep `coveralls` ? Hence, why this PR is not getting a response from the coveralls bot...

**Comment by pelson** [comment]
Great. Coverage has increased by 2.8%. 👍

**Comment by pelson** [comment]
Thanks @bjlittle.

**Pull request merged**